src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -152,21 +154,21 @@ HostSymbolStartAnalysisScope getSymbolAnalysisScopeCore()
                             var dependentSymbols = getDependentSymbols();
                             lock (_gate)
                             {
-                                _lazyPendingMemberSymbolsMapOpt = _lazyPendingMemberSymbolsMapOpt ?? new Dictionary<ISymbol, HashSet<ISymbol>>();
+                                _lazyPendingMemberSymbolsMap ??= new Dictionary<ISymbol, HashSet<ISymbol>?>();
```

Comment:
📝 This one only needed the `?` added

Reply:
No, the entry here could have null value for non-namespace or type symbols, which are likely the most common entries here and we would like to avoid hashset allocation for them. If there is potential to improve performance using other data structure type, we can do it as a follow-up change with performance measurements/improvements being verified.